Yosemite/YosemiteTests/Stores/OrderStoreTests+FetchFilteredAndAllOrders.swift

```diff
@@ -0,0 +1,204 @@
+        network.simulateResponse(requestUrlSuffix: "orders", filename: Fixtures.ordersLoadAllJSON.fileName)
+        network.simulateResponse(requestUrlSuffix: "orders", filename: Fixtures.ordersLoadAll2JSON.fileName)
+
+        // Act
+        executeActionAndWait(using: createOrderStore(using: network),
+                             statusKey: OrderStatusEnum.completed.rawValue,
+                             deleteAllBeforeSaving: true)
+
+        // Assert
+        XCTAssertEqual(countOrders(),
+                       Fixtures.ordersLoadAllJSON.ordersCount + Fixtures.ordersLoadAll2JSON.ordersCount)
+    }
+
+    func testWhenNotGivenAFilterItFetchesTheAllOrdersListOnly() {
+        // Arrange
+        let network = MockupNetwork(useResponseQueue: true)
+        network.simulateResponse(requestUrlSuffix: "orders", filename: Fixtures.ordersLoadAllJSON.fileName)
+        network.simulateResponse(requestUrlSuffix: "orders", filename: Fixtures.ordersLoadAll2JSON.fileName)
+
+        // Act
+        executeActionAndWait(using: createOrderStore(using: network),
+                             statusKey: nil,
+                             deleteAllBeforeSaving: true)
+
+        // Assert
+        XCTAssertEqual(countOrders(), Fixtures.ordersLoadAllJSON.ordersCount)
+    }
+
+    func testWhenAnErrorHappensItWillNotDeleteAllOrders() {
+        // Arrange
+        insert(order: Fixtures.order)
+
+        let network = MockupNetwork()
+        network.simulateError(requestUrlSuffix: "orders", error: NetworkError.timeout)
+
+        // Act
+        executeActionAndWait(using: createOrderStore(using: network),
+                             statusKey: OrderStatusEnum.processing.rawValue,
+                             deleteAllBeforeSaving: true)
+
+        // Assert
+        // The previously saved order should still exist
+        XCTAssertNotNil(findOrder(withID: Fixtures.order.orderID))
+        XCTAssertEqual(countOrders(), 1)
+    }
+}
+
+// MARK: - Private
+
+private extension OrderStoreTests_FetchFilteredAndAllOrders {
+    func createOrderStore(using network: Network) -> OrderStore {
+        OrderStore(dispatcher: Dispatcher(), storageManager: storageManager, network: network)
+    }
+
+    func executeActionAndWait(using store: OrderStore, statusKey: String?, deleteAllBeforeSaving: Bool) {
+        let expectation = self.expectation(description: "fetch")
+
+        let action = OrderAction.fetchFilteredAndAllOrders(
+            siteID: Fixtures.siteID,
+            statusKey: statusKey,
+            deleteAllBeforeSaving: deleteAllBeforeSaving,
+            pageSize: 50) { _ in
+                expectation.fulfill()
+        }
+
+        store.onAction(action)
+
+        wait(for: [expectation], timeout: Constants.expectationTimeout)
+    }
+
+    func insert(order: Networking.Order) {
+        let storageOrder = viewStorage.insertNewObject(ofType: Storage.Order.self)
+        storageOrder.update(with: Fixtures.order)
```

Comment:
perhaps we can add this helper to [`MockupStorage+Sample.swift`](https://github.com/woocommerce/woocommerce-ios/blob/develop/Yosemite/YosemiteTests/Mockups/MockupStorage+Sample.swift)?

Reply:
Ooohh. Didn't know that that existed. 

Do you mean the `Fixtures.order` or the `insert()` function?